src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+            SET_CUR_FORM_BTN_SLOT(btn, SLOT_NONE);
+        }
+    }
+}
+
+void Inventory_UnequipItem(s16 item) {
+    s16 btn;
+
+    for (btn = 1; btn < 4; btn++) {
+        if (GET_CUR_FORM_BTN_ITEM(btn) == item) {
+            SET_CUR_FORM_BTN_ITEM(btn, ITEM_NONE);
+            SET_CUR_FORM_BTN_SLOT(btn, SLOT_NONE);
+        }
+    }
+}
+
+s32 Inventory_ReplaceItem(GlobalContext* globalCtx, u8 oldItem, u8 newItem) {
+    u8 i;
+
+    for (i = 0; i < 24; i++) {
+        if (gSaveContext.save.inventory.items[i] == oldItem) {
+            gSaveContext.save.inventory.items[i] = newItem;
+
+            for (i = 1; i < 4; i++) {
+                if (GET_CUR_FORM_BTN_ITEM(i) == oldItem) {
+                    SET_CUR_FORM_BTN_ITEM(i, newItem);
+                    Interface_LoadItemIconImpl(globalCtx, i);
+                    break;
+                }
+            }
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateDeitySwordEquip(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    u8 btn;
+
+    if (CUR_FORM == PLAYER_FORM_FIERCE_DEITY) {
+        interfaceCtx->unk_21C = 0;
+        interfaceCtx->unk_21E = 0;
+        if ((((gSaveContext.save.playerForm > 0) && (gSaveContext.save.playerForm < 4))
+                 ? 1
+                 : gSaveContext.save.playerForm >> 1) == 0) {
+            BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_SWORD_DEITY;
+        } else {
+            if (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) == ITEM_SWORD_DEITY) {
+                if (GET_CUR_EQUIP_VALUE(EQUIP_SWORD) == 0) {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_NONE;
+                } else {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) =
+                        GET_CUR_EQUIP_VALUE(EQUIP_SWORD) + ITEM_SWORD_KOKIRI - 1;
+                }
+            }
+        }
+    }
+
+    for (btn = 0; btn < 1; btn++) {
+        if ((GET_CUR_FORM_BTN_ITEM(btn) != ITEM_NONE) && (GET_CUR_FORM_BTN_ITEM(btn) != ITEM_UNK_FD)) {
+            Interface_LoadItemIconImpl(globalCtx, btn);
+        }
+    }
+}
+
+s32 Inventory_HasEmptyBottle(void) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == ITEM_BOTTLE) {
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 Inventory_HasItemInBottle(u8 item) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == item) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateBottleItem(GlobalContext* globalCtx, u8 item, u8 btn) {
+    gSaveContext.save.inventory.items[GET_CUR_FORM_BTN_SLOT(btn)] = item;
+    SET_CUR_FORM_BTN_ITEM(btn, item);
+
+    Interface_LoadItemIconImpl(globalCtx, btn);
+
+    globalCtx->pauseCtx.cursorItem[0] = item;
```

Comment:
PAUSE_0?

Reply:
Ugh, `z_file_choose.h` was put in a very local place inside `mm/src/overlays/gamestates/ovl_file_choose/z_file_choose.h` instead of just `include/`, should I move this header file?

Reply:
I don't see the relevance. Also, it's an overlay header like any other, so I think it should stay where it is.